File: app/src/main/java/com/example/learningtolive/CountryActivity.java
package com.example.learningtolive;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class CountryActivity extends AppCompatActivity {
    String country;
    Button dailyLifeButton;
    Button healthButton;
    Button settlingButton;
    Button migrantButton;
    Button languageButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            country = extras.getString("country");
        }
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_country);
        TextView countryNameTV = (TextView) findViewById(R.id.countryNameTV);
        countryNameTV.setText(country);

        dailyLifeButton = (Button) findViewById(R.id.daily_life_button);
        dailyLifeButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Category c = new Category(Category.Name.DAILYLIFE);
                sendCategory(v, c);
            }
        });

        healthButton = (Button) findViewById(R.id.health_button);
        healthButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Category c = new Category(Category.Name.HEALTH);
                sendCategory(v, c);
            }
        });

        settlingButton = (Button) findViewById(R.id.settling_button);
        settlingButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Category c = new Category(Category.Name.SETTLINGIN);
                sendCategory(v, c);
            }
        });

        migrantButton = (Button) findViewById(R.id.migrant_button);
        migrantButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Category c = new Category(Category.Name.MIGRANTSTATUS);
                sendCategory(v, c);
            }
        });

        languageButton = (Button) findViewById(R.id.language_button);
        languageButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Category c = new Category(Category.Name.LANGUAGE);
                sendCategory(v, c);
            }
        });
    }

    private void sendCategory(View view, Category category) {
        Intent intent = new Intent(this, CategoryActivity.class);
        intent.putExtra("category", category);
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/learningtolive/CountryActivity.kt
package com.example.learningtolive

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.learningtolive.com.example.kotlinmultiplatformsharedmodule.Category

class CountryActivity : AppCompatActivity() {
    lateinit var country: String
    private lateinit var dailyLifeButton: Button
    private lateinit var healthButton: Button
    private lateinit var settlingButton: Button
    private lateinit var migrantButton: Button
    private lateinit var languageButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        val extras = intent.extras
        if (extras != null) {
            country = extras.getString("country")!!
        }
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_country)

        val countryNameTV = findViewById<View>(R.id.countryNameTV) as TextView
        countryNameTV.text = country

        dailyLifeButton = findViewById<View>(R.id.daily_life_button) as Button
        dailyLifeButton.setOnClickListener { v ->
            val c = Category(Category.Name.DAILYLIFE)
            sendCategory(v, c, country)
        }
        healthButton = findViewById<View>(R.id.health_button) as Button
        healthButton.setOnClickListener { v ->
            val c = Category(Category.Name.HEALTH)
            sendCategory(v, c, country)
        }
        settlingButton = findViewById<View>(R.id.settling_button) as Button
        settlingButton.setOnClickListener { v ->
            val c = Category(Category.Name.SETTLINGIN)
            sendCategory(v, c, country)
        }
        migrantButton = findViewById<View>(R.id.migrant_button) as Button
        migrantButton.setOnClickListener { v ->
            val c = Category(Category.Name.MIGRANTSTATUS)
            sendCategory(v, c, country)
        }
        languageButton = findViewById<View>(R.id.language_button) as Button
        languageButton.setOnClickListener { v ->
            val c = Category(Category.Name.LANGUAGE)
            sendCategory(v, c, country)
        }
    }

    private fun sendCategory(view: View, category: Category, country: String) {
        val intent = Intent(this, CategoryActivity::class.java)
        intent.putExtra("category", category)
        intent.putExtra("country", country)
        startActivity(intent)
    }
}
